src/main/java/org/folio/service/event/DomainEventPublisher.java

```diff
@@ -48,20 +48,23 @@ public Future<Void> publish(K key, DomainEvent<T> event, Map<String, String> oka
         .topic(kafkaTopic)
         .propagateOkapiHeaders(okapiHeaders)
         .build();
-    log.info("publish:: kafkaRecord = [{}]", producerRecord);
 
     KafkaProducer<K, String> producer = getOrCreateProducer();
     log.info("publish:: Producer created, sending the record...");
 
     return producer.send(producerRecord)
-      .onSuccess(r -> log.info("publish:: Succeeded sending domain event with key [{}], " +
-        "kafka record [{}]", key, producerRecord))
-      .<Void>mapEmpty()
-      .onFailure(cause -> {
-        log.error("publish:: Unable to send domain event with key [{}], kafka record [{}]",
-          key, producerRecord, cause);
-        failureHandler.handle(cause, producerRecord);
-      });
+      .onComplete(r -> {
+        producer.end(ear -> producer.close());
```

Comment:
Missing closing producer will cause memory leak like this: https://folio-org.atlassian.net/browse/MODINVSTOR-794
All other places make closing of kafka producer: 
https://github.com/search?q=org%3Afolio-org+producer.close%28%29&type=code